mlrun/api/db/sqldb/db.py

```diff
@@ -1473,9 +1473,38 @@ def patch_feature_set(
             always_overwrite=True,
         )
 
-    def delete_feature_set(self, session, project, name):
-        self._delete(session, FeatureSet.Tag, project=project, obj_name=name)
-        self._delete(session, FeatureSet, project=project, name=name)
+    def _delete_feature_store_object(self, session, cls, project, name, tag, uid):
+        if tag and uid:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Both uid and tag specified when deleting an object."
+            )
+
+        object_id = None
+        if uid:
+            object_record = self._query(
+                session, cls, project=project, name=name, uid=uid
+            ).one_or_none()
+            if object_record is None:
+                return
+            object_id = object_record.id
+        elif tag:
+            tag_record = self._query(
+                session, cls.Tag, project=project, name=tag, obj_name=name
+            ).one_or_none()
+            if tag_record is None:
+                return
+            object_id = tag_record.obj_id
+
+        if object_id:
+            self._delete(session, cls, id=object_id)
+            self._delete(session, cls.Tag, obj_id=object_id)
```

Comment:
If you got tag, not sure you need to remove the object, it might be referenced by other tag (if it is, just need to remove the tag IMO)

Reply:
That's not my view of it - this is a call to remove FeatureSet/FeatureVector, and therefore should delete the specific object version that is referenced. At some point we should create a tag endpoint that will handle deletion of tags, but I don't think this is what needs to happen here. 
To stress the point - the `reference` passed (which happens to be a tag) is only used to find the specific object instance (or REST resource). It should not modify the decision on what to delete - once an object instance is found, it should be deleted.